public/app/features/explore/ReturnToDashboardButton.tsx

```diff
@@ -0,0 +1,108 @@
+
+import kbn from '../../core/utils/kbn';
+import { getDashboardSrv } from '../dashboard/services/DashboardSrv';
+import { StoreState } from 'app/types';
+import { ExploreId } from 'app/types/explore';
+import { updateLocation } from 'app/core/actions';
+import { setDashboardQueriesToUpdateOnLoad } from '../dashboard/state/reducers';
+
+const { ButtonSelect } = LegacyForms;
+
+interface Props {
+  originPanelId: number;
+  exploreId: ExploreId;
+  queries: DataQuery[];
+  updateLocation: typeof updateLocation;
+  setDashboardQueriesToUpdateOnLoad: typeof setDashboardQueriesToUpdateOnLoad;
+}
+
+export const UnconnectedReturnToDashboardButton: FC<Props> = ({
+  originPanelId,
+  updateLocation,
+  setDashboardQueriesToUpdateOnLoad,
+  queries,
+}) => {
+  const originDashboardIsEditable = originPanelId && Number.isInteger(originPanelId);
+  const panelReturnClasses = classNames('btn', 'navbar-button', {
+    'btn--radius-right-0': originDashboardIsEditable,
+    'navbar-button navbar-button--border-right-0': originDashboardIsEditable,
+  });
+
+  const cleanQueries = (queries: DataQuery[]) => {
+    return queries.map((query: DataQuery & { context?: string }) => {
+      delete query.context;
+      delete query.key;
+      return query;
+    });
+  };
+
+  const returnToPanel = async ({ withChanges = false } = {}) => {
+    const dashboardSrv = getDashboardSrv();
+    const dash = dashboardSrv.getCurrent();
+    const titleSlug = kbn.slugifyForUrl(dash.title);
+
+    if (withChanges) {
+      setDashboardQueriesToUpdateOnLoad({
+        panelId: originPanelId!,
+        queries: cleanQueries(queries),
+      });
+    }
+
+    const query: any = {};
+
+    if (withChanges || dash.panelInEdit) {
+      query.editPanel = originPanelId;
+    } else if (dash.panelInView) {
+      query.viewPanel = originPanelId;
+    }
+
+    updateLocation({ path: `/d/${dash.uid}/:${titleSlug}`, query });
+  };
+
+  return (
+    <div className="explore-toolbar-content-item">
+      <Tooltip content={'Return to panel'} placement="bottom">
+        <button
+          data-testid="returnButton"
+          title={'Return to panel'}
+          className={panelReturnClasses}
+          onClick={() => returnToPanel()}
+        >
+          <Icon name="arrow-left" />
+        </button>
+      </Tooltip>
+      {originDashboardIsEditable && (
+        <div data-testid="returnButtonWithChanges">
+          <ButtonSelect
+            className="navbar-button--attached btn--radius-left-0$"
+            options={[{ label: 'Return to panel with changes', value: '' }]}
+            onChange={() => returnToPanel({ withChanges: true })}
+            maxMenuHeight={380}
+          />
+        </div>
+      )}
+    </div>
+  );
+};
+
+function mapStateToProps(state: StoreState, { exploreId }: { exploreId: ExploreId }) {
+  const explore = state.explore;
+  const { datasourceInstance, queries } = explore[exploreId];
+  return {
+    exploreId,
+    datasourceInstance,
+    queries,
```

Comment:
nit: we could probably get also the originalPanelId from redux instead of passing it as prop and then just not render anything here if there isn't any value basically moving this condition `originPanelId && Number.isInteger(originPanelId) && !splitted` inside the component.

Reply:
Moved https://github.com/grafana/grafana/pull/29752/commits/94eea0f47c81608a34af4874d990223ffde13347. I have also added a test for the scenario with split mode.